apis/v1alpha1/gateway_types.go

```diff
@@ -53,6 +53,8 @@ type GatewayList struct {
 type GatewaySpec struct {
 	// GatewayClassName used for this Gateway. This is the name of a
 	// GatewayClass resource.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=100
```

Comment:
is 100 a common limit for this?

DNS can be 253
Maybe 255 or so

Reply:
Thanks for catching this! This should have been 253 because it's referring to the name of a Kubernetes resource. Most Kubernetes resources use the standard `IsDNS1123Subdomain` validation for names which includes a [limit of 253 chars](https://github.com/kubernetes/kubernetes/blob/4373d9aadd7b98aace0ed8b90ff45b255d4b1e88/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L202). Should be updated now.